### ISSUE TITLE
Consider making API generic

### ISSUE THREAD

**Issue by adamlofts** [adamlofts/leveldb_dart#3]
Would be nice to have a `LevelIterator<K, V>`. I think this requires either

(1) Generic methods: https://chromium.googlesource.com/external/github.com/dart-lang/dev_compiler/+/refs/heads/master/doc/GENERIC_METHODS.md
or
(2) Making `LevelDB<K, V>`

Presumably if (2) then keyEncoding and valueEncoding would move into the LevelDB constructor. This would restrict DB updates to one encoding.

**Comment by adamlofts** [comment]
Working on `generic` branch. Waiting on resolution of https://github.com/dart-lang/sdk/issues/29349

**Comment by adamlofts** [comment]
Fixed in 9cd851f
